homeassistant/components/hydrawise/coordinator.py

```diff
@@ -12,59 +11,102 @@
 from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
 from homeassistant.util.dt import now
 
-from .const import DOMAIN, LOGGER
+from .const import DOMAIN, LOGGER, MAIN_SCAN_INTERVAL, WATER_USE_SCAN_INTERVAL
 
 
 @dataclass
 class HydrawiseData:
     """Container for data fetched from the Hydrawise API."""
 
     user: User
-    controllers: dict[int, Controller]
-    zones: dict[int, Zone]
-    sensors: dict[int, Sensor]
-    daily_water_summary: dict[int, ControllerWaterUseSummary]
+    controllers: dict[int, Controller] = field(default_factory=dict)
+    zones: dict[int, Zone] = field(default_factory=dict)
+    sensors: dict[int, Sensor] = field(default_factory=dict)
+    daily_water_summary: dict[int, ControllerWaterUseSummary] = field(
+        default_factory=dict
+    )
+
+
+@dataclass
+class HydrawiseUpdateCoordinators:
+    """Container for all Hydrawise DataUpdateCoordinator instances."""
+
+    main: HydrawiseMainDataUpdateCoordinator
+    water_use: HydrawiseWaterUseDataUpdateCoordinator
 
 
 class HydrawiseDataUpdateCoordinator(DataUpdateCoordinator[HydrawiseData]):
-    """The Hydrawise Data Update Coordinator."""
+    """Base class for Hydrawise Data Update Coordinators."""
 
     api: Hydrawise
 
-    def __init__(
-        self, hass: HomeAssistant, api: Hydrawise, scan_interval: timedelta
-    ) -> None:
+
```

Comment:
Does having multiple coordinators really simplify the code? It feels like we are just adding a lot of code that wasn't really necessary in your previous version.

Reply:
It's not about simplification, it's about correctness. As @frenck pointed out earlier, conditionally fetching data breaks assumptions about what the `homeassistant.update_entity` service call is supposed to do. For example, if you call `homeassistant.update_entity` wanting to get fresh data about a flow sensor, it won't do anything if we're not always pulling data in every coordinator refresh.